### PR TITLE
RP2040: Update core; add mDNS support

### PR DESCRIPTION
**Pull request by GUVWAF**
Updated the core, hopefully to fix potential FreeRTOS issues. 
It also includes a new `SimpleMDNS` library that does work with FreeRTOS. Added a "meshtastic" service as there is no HTTP/HTTPS service, which the Android app now also looks for: https://github.com/meshtastic/Meshtastic-Android/pull/1401. Tested this and it indeed finds the IP address now.

Lastly, added the `-g` option to link the source files for debugging.

### CODE REVIEW

**Review by fifieldt — APPROVED** [review]
Checked that it only affects rpi, changes seem fine.

File: src/mesh/wifi/WiFiAPClient.cpp [hunk 1]
```diff
@@ -59,19 +61,21 @@ static void onNetworkConnected()
...
         } else {
-            LOG_INFO("mDNS responder started");
             LOG_INFO("mDNS Host: Meshtastic.local");
+#ifdef ARCH_ESP32
             MDNS.addService("http", "tcp", 80);
             MDNS.addService("https", "tcp", 443);
-        }
-#else // ESP32 handles this in WiFiEvent
-        LOG_INFO("Obtained IP address: %s", WiFi.localIP().toString().c_str());
+#else
```
Comment (fifieldt) [inline src/mesh/wifi/WiFiAPClient.cpp:72]: is it worth guarding this in case other architectures have wifi in the future?

**Review by fifieldt — APPROVED** [review]

**Comment by broglep** [comment]
Great to see MDNS support for the actual meshtastic tcp port, wanted to suggest that as well. 
 Is there any particular reason why not always announcing `MDNS.addService("meshtastic", "tcp", 4403);`, instead just for `ARCH_RP2040` ?

**Comment by GUVWAF** [comment]
Good question, it seemed a bit redundant to me, but it would indeed be more correct.

**Pull request merged**